docker-entrypoint/docker-entrypoint.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	//For testing purposes
+	//"k8s.io/kubernetes/pkg/client/restclient"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	WAITING  = "waiting"
+	READY    = "ready"
+	RUNNING  = "Running"
+	INTERVAL = 2
+)
+
+var (
+	//"Info logger""
+	Info *log.Logger
+	//"Error logger"
+	Error *log.Logger
+)
+
+type dependency interface {
+	Exists(namespace string, name string) error
+	DepResolved(namespace string, name string) (bool, error)
+	GetType() string
+}
+
+func WaitFor(dep dependency, namespace string, names []string) error {
+	depState := WAITING
+	if names == nil {
+		depState = READY
+		Info.Println("Container has no ", dep.GetType(), " dependencies")
+	}
+
+	for depState == WAITING {
+
+		for _, name := range names {
+
+			name := strings.TrimSpace(name)
+			err := dep.Exists(namespace, name)
+			if err != nil {
+				Info.Println(dep.GetType(), name, " doesn't exists -> State waiting")
+				depState = WAITING
+				break
+			}
+			res, err := dep.DepResolved(namespace, name)
```

Comment:
Why do we test both dep.Exists and dep.DepResolved? Is there any technical usage of having and exposing both functions in the interface? Even the info message is the same.